src/main/java/com/scalar/db/storage/cosmos/TableMetadataHandler.java

```diff
@@ -0,0 +1,52 @@
+package com.scalar.db.storage.cosmos;
+
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.models.PartitionKey;
+import com.scalar.db.api.Operation;
+import com.scalar.db.exception.storage.StorageRuntimeException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A handler to read and cache {@link TableMetadata} to know the type of each column
+ *
+ * @author Yuji Ito
+ */
+@ThreadSafe
+public class TableMetadataHandler {
```

Comment:
Could be my preference but 
```suggestion
public class TableMetadataManager {
```
might be a better naming since it's more like managing table metadata instead of handling.